cmd/api/handlers-nodes.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/jmpsec/osctrl/pkg/settings"
+	"github.com/jmpsec/osctrl/pkg/utils"
+)
+
+// GET Handler for single JSON nodes
+func apiNodeHandler(w http.ResponseWriter, r *http.Request) {
+	incMetric(metricAPIReq)
+	utils.DebugHTTPDump(r, settingsmgr.DebugHTTP(settings.ServiceAPI), false)
+	vars := mux.Vars(r)
+	// Extract uuid
+	uuid, ok := vars["uuid"]
+	if !ok {
+		incMetric(metricAPIErr)
+		apiErrorResponse(w, "error getting uuid", nil)
+		return
+	}
+	// Get node by UUID
+	node, err := nodesmgr.GetByUUID(uuid)
+	if err != nil {
+		incMetric(metricAPIErr)
+		if err.Error() == "record not found" {
+			log.Printf("node not found: %s", uuid)
+			apiHTTPResponse(w, JSONApplicationUTF8, http.StatusNotFound, ApiErrorResponse{Error: "node not found"})
```

Comment:
Is what this returns a similar JSON format to successful run? If so that'll make it easier to parse in prospective clients.

Reply:
When the requested node exists, it does return the structure for `OsqueryNode{}`, with all its fields in it. Does it make more sense to return always a JSON response with one field for message, and another for data?

Reply:
No that should be fine, a client can easily check that the fields are empty :)